config/src/config/network_config.rs

```diff
@@ -50,6 +50,7 @@ pub const MAX_APPLICATION_MESSAGE_SIZE: usize =
     (MAX_MESSAGE_SIZE - MAX_MESSAGE_METADATA_SIZE) - MESSAGE_PADDING_SIZE; /* The message size that applications should check against */
 pub const MAX_FRAME_SIZE: usize = 4 * 1024 * 1024; /* 4 MiB large messages will be chunked into multiple frames and streamed */
 pub const MAX_MESSAGE_SIZE: usize = 64 * 1024 * 1024; /* 64 MiB */
+pub const MAX_PARALLEL_DESERIALIZATION_TASKS: usize = 500; // This is per application and should be enough for max load
```

Comment:
Probably should benchmark / it should be roughly same as the number of cores

Reply:
Yeah, I've changed this to default to the number of CPU cores and added a test 😄